### PR TITLE
[WIP] bump(main/screen): 5.0.0

### PR DESCRIPTION
**Pull request by TomJo2000**
I tried updating the `screen` package.

I would describe the experience as very similar to eating glass.
The tarball has a different layout from the repo.
Half the stuff we patch seems to just be gone in 5.0.0
and it seems like `screen` now depends on `pam` for authentication so that will need to be patched.

I've inflicted enough torture on myself for one day, so I'll leave this here for anyone interested in making `screen` 5.0.0 work.

<details><summary>Here's the current build error.</summary>
<p>

```styl
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building screen for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13134  100 13134    0     0  19777      0 --:--:-- --:--:-- --:--:-- 19780
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   833  100   833    0     0   1304      0 --:--:-- --:--:-- --:--:--  1303
gpg: Signature made Sat 21 Dec 2024 02:02:18 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13320  100 13320    0     0  20595      0 --:--:-- --:--:-- --:--:-- 20587
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   833  100   833    0     0   1294      0 --:--:-- --:--:-- --:--:--  1295
gpg: Signature made Tue 17 Dec 2024 06:32:46 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13120  100 13120    0     0  20382      0 --:--:-- --:--:-- --:--:-- 20372
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   833  100   833    0     0   1302      0 --:--:-- --:--:-- --:--:--  1301
gpg: Signature made Sat 21 Dec 2024 01:42:03 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading dependency ca-certificates@1:2024.09.24 if necessary...
Skipping already built dependency ca-certificates@1:2024.09.24
Downloading dependency ca-certificates-java@1:2024.09.24 if necessary...
Skipping already built dependency ca-certificates-java@1:2024.09.24
Downloading dependency zlib@1.3.1 if necessary...
Skipping already built dependency zlib@1.3.1
Downloading dependency openssl@1:3.3.2 if necessary...
Skipping already built dependency openssl@1:3.3.2
Downloading dependency openssl-tool@1:3.3.2 if necessary...
Skipping already built dependency openssl-tool@1:3.3.2
Downloading dependency libcrypt@0.2-5 if necessary...
Skipping already built dependency libcrypt@0.2-5
Downloading dependency ncurses@6.5.20240831-2 if necessary...
Skipping already built dependency ncurses@6.5.20240831-2
Downloading dependency ncurses-utils@6.5.20240831-2 if necessary...
Skipping already built dependency ncurses-utils@6.5.20240831-2
Downloading dependency ncurses-ui-libs@6.5.20240831-2 if necessary...
Skipping already built dependency ncurses-ui-libs@6.5.20240831-2
Downloading dependency ncurses-ui-libs-static@6.5.20240831-2 if necessary...
Skipping already built dependency ncurses-ui-libs-static@6.5.20240831-2
Applying patch: fileio.c.patch
Applying patch: misc.c.patch
Applying patch: no-setuid.patch
Applying patch: os.h.patch
Applying patch: screen.h.patch
Applying patch: utmp.c.patch
Applying patch: window.c.patch
configure: WARNING: unrecognized options: --disable-dependency-tracking, --disable-rpath, --disable-rpath-hack, --enable-colors256, --disable-nls, --enable-shared, --enable-static
checking for aarch64-linux-android-gcc... aarch64-linux-android-clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether aarch64-linux-android-clang accepts -g... yes
checking for aarch64-linux-android-clang option to enable C11 features... none needed
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for aarch64-linux-android-gcc... (cached) aarch64-linux-android-clang
checking whether the compiler supports GNU C... (cached) yes
checking whether aarch64-linux-android-clang accepts -g... (cached) yes
checking for aarch64-linux-android-clang option to enable C11 features... (cached) none needed
checking for gawk... gawk
checking for a BSD-compatible install... /usr/bin/install -c
checking for execvpe... yes
checking for seteuid... yes
checking for setegid... yes
checking for setreuid... yes
checking for setresuid... yes
checking for library containing tgetent... -lcurses
checking for library containing crypt... none required
checking for pty.h... yes
checking for util.h... no
checking for libutil.h... no
checking for library containing openpty... none required
checking for library containing socket... none required
checking for langinfo.h... yes
checking for dirent.h... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating config.h
configure: WARNING: unrecognized options: --disable-dependency-tracking, --disable-rpath, --disable-rpath-hack, --enable-colors256, --disable-nls, --enable-shared, --enable-static

Configuration:

 PAM support: .............................. no
 telnet support: ........................... no
 utmp support: ............................. no
 global socket directory: .................. no

 system screenrc location: ................. /etc/screenrc
 pty mode: ................................. 0622
 pty group: ................................ 5
 pty on read only file system: ............. no

!!! WARNING !!!
YOU ARE DISABLING PAM SUPPORT!
FOR screen TO WORK IT WILL NEED TO RUN AS SUID root BINARY
THIS CONFIGURATION IS _HIGHLY_ NOT RECOMMENDED!
!!! WARNING !!!
AWK=gawk srcdir=/home/builder/.termux-build/screen/src sh /home/builder/.termux-build/screen/src/term.sh
aarch64-linux-android-clang -c -I/data/data/com.termux/files/usr/include -iquote. -DSCREENENCODINGS='"/data/data/com.termux/files/usr/share/screen/utf8encodings"' -DBUILD_DATE='"2024-12-21 13:12:04"' -fstack-protector-strong -Oz -DGETUTENT -Wall -Wextra -std=c17 /home/builder/.termux-build/screen/src/winmsgcond.c -o winmsgcond.o
aarch64-linux-android-clang -c -I/data/data/com.termux/files/usr/include -iquote. -DSCREENENCODINGS='"/data/data/com.termux/files/usr/share/screen/utf8encodings"' -DBUILD_DATE='"2024-12-21 13:12:04"' -fstack-protector-strong -Oz -DGETUTENT -Wall -Wextra -std=c17 /home/builder/.termux-build/screen/src/setuid.c -o setuid.o
AWK=gawk CC="aarch64-linux-android-clang -fstack-protector-strong -Oz -DGETUTENT -Wall -Wextra -std=c17" srcdir=/home/builder/.termux-build/screen/src sh /home/builder/.termux-build/screen/src/comm.sh
aarch64-linux-android-clang -c -I/data/data/com.termux/files/usr/include -iquote. -DSCREENENCODINGS='"/data/data/com.termux/files/usr/share/screen/utf8encodings"' -DBUILD_DATE='"2024-12-21 13:12:04"' -fstack-protector-strong -Oz -DGETUTENT -Wall -Wextra -std=c17 kmapdef.c -o kmapdef.o
aarch64-linux-android-clang -c -I/data/data/com.termux/files/usr/include -iquote. -DSCREENENCODINGS='"/data/data/com.termux/files/usr/share/screen/utf8encodings"' -DBUILD_DATE='"2024-12-21 13:12:04"' -fstack-protector-strong -Oz -DGETUTENT -Wall -Wextra -std=c17 /home/builder/.termux-build/screen/src/term.c -o term.o
/home/builder/.termux-build/screen/src/setuid.c:3:18: warning: unused parameter 'uid' [-Wunused-parameter]
    3 | int setuid(uid_t uid) {
      |                  ^
/home/builder/.termux-build/screen/src/setuid.c:7:18: warning: unused parameter 'gid' [-Wunused-parameter]
    7 | int setgid(gid_t gid) {
      |                  ^
/home/builder/.termux-build/screen/src/setuid.c:11:19: warning: unused parameter 'gid' [-Wunused-parameter]
   11 | int seteuid(gid_t gid) {
      |                   ^
/home/builder/.termux-build/screen/src/setuid.c:15:19: warning: unused parameter 'gid' [-Wunused-parameter]
   15 | int setegid(gid_t gid) {
      |                   ^
/home/builder/.termux-build/screen/src/setuid.c:19:21: warning: unused parameter 'ruid' [-Wunused-parameter]
   19 | int setresuid(uid_t ruid, uid_t euid, uid_t suid) {
      |                     ^
/home/builder/.termux-build/screen/src/setuid.c:19:33: warning: unused parameter 'euid' [-Wunused-parameter]
   19 | int setresuid(uid_t ruid, uid_t euid, uid_t suid) {
      |                                 ^
/home/builder/.termux-build/screen/src/setuid.c:19:45: warning: unused parameter 'suid' [-Wunused-parameter]
   19 | int setresuid(uid_t ruid, uid_t euid, uid_t suid) {
      |                                             ^
/home/builder/.termux-build/screen/src/setuid.c:23:21: warning: unused parameter 'rgid' [-Wunused-parameter]
   23 | int setresgid(gid_t rgid, gid_t egid, gid_t sgid) {
      |                     ^
/home/builder/.termux-build/screen/src/setuid.c:23:33: warning: unused parameter 'egid' [-Wunused-parameter]
   23 | int setresgid(gid_t rgid, gid_t egid, gid_t sgid) {
      |                                 ^
/home/builder/.termux-build/screen/src/setuid.c:23:45: warning: unused parameter 'sgid' [-Wunused-parameter]
   23 | int setresgid(gid_t rgid, gid_t egid, gid_t sgid) {
      |                                             ^
kmapdef.c:5:10: fatal error: 'kmapdef.h' file not found
    5 | #include "kmapdef.h"
      |          ^~~~~~~~~~~
1 error generated.
10 warnings generated.
make: *** [Makefile:69: kmapdef.o] Error 1
make: *** Waiting for unfinished jobs....
```

</p>
</details>

In conclusion, abandon all hope ye who enter here.
If somebody tries to force me to work on `screen` again I will gladly take a plate of broken glass instead.

### CODE REVIEW

**Comment by Biswa96** [comment]
Could you try with `TERMUX_PKG_BUILD_IN_SRC=true` please?

**Comment by TomJo2000** [comment]
You know, after taking 20d6 free psychic damage, that hadn't occurred to me before.

Still fails, but looks like it's for a different reason now.

**Comment by TomJo2000** [comment]
Thanks.
Let's hope it actually works on device.

Though to be honest if I never have to touch this package again it'll be too soon.

**Comment by robertkirkman** [comment]
I also made one but now I see someone else made one at the same time, 
they are very, very similar (we independently made almost identical changes simultaneously) but yours is better than mine so that's good,
When I compare them, there is only one comment I have which is, `-DGETUTENT` definitely does not do anything anymore, but it being there is harmless so it doesn't matter.
